streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -184,8 +184,7 @@ class CheckpointWriter(
   val executor = Executors.newFixedThreadPool(1)
   val compressionCodec = CompressionCodec.createCodec(conf)
   private var stopped = false
-  private var _fs: FileSystem = _
-
+  @volatile private[this] var fs: FileSystem = null
```

Comment:
It's actually referenced by inner classes. I think the idea here is that we'll avoid calling `.getFileSystem` too much by caching the result, but will have the ability to clear the cache in case the cached FileSystem seems to be in a bad state.